From 15d721925e0e6452fdd6fa709e31cf3064c1a65d Mon Sep 17 00:00:00 2001
From: Saravanakumar Duraisamy <quic_saradura@quicinc.com>
Date: Mon, 7 Aug 2023 15:35:08 +0530
Subject: [PATCH] ath11k: Removed unwanted rcu_read_lock which cause rcu stall

With 2.4G radio on 6.1 kernel when reboot/wifi down/ Wifi up rcu stall observed.
When we checked the RCU trace "RCU nest depth: -1" found which means unlock taken
without lock. During the issue case double unlock happend in "fw_stats_data_parse".

Hence removing the unwanted rcu_read_unlock.

Signed-off-by: Saravanakumar Duraisamy <quic_saradura@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/wmi.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath11k/wmi.c b/drivers/net/wireless/ath/ath11k/wmi.c
index 4cc284e..0124900 100644
--- a/drivers/net/wireless/ath/ath11k/wmi.c
+++ b/drivers/net/wireless/ath/ath11k/wmi.c
@@ -7451,7 +7451,6 @@ static int ath11k_wmi_tlv_fw_stats_data_parse(struct ath11k_base *ab,
 		}
 		spin_unlock_bh(&ar->data_lock);
 	}
-	rcu_read_unlock();
 
 	for (i = 0; i < ev->num_vdev_stats; i++) {
 		const struct wmi_vdev_stats *src;
-- 
2.17.1

